pkg/operator/ceohelpers/bootstrap.go

```diff
@@ -196,5 +196,17 @@ func IsBootstrapComplete(configMapClient corev1listers.ConfigMapLister, staticPo
 			return false, nil
 		}
 	}
+
+	// check if etcd-bootstrap member is still present within the etcd cluster membership
+	membersList, err := etcdClient.MemberList(context.Background())
+	if err != nil {
+		return false, fmt.Errorf("IsBootstrapComplete couldn't list the etcd cluster members: %w", err)
+	}
+	for _, m := range membersList {
+		if m.Name == "etcd-bootstrap" {
+			return false, fmt.Errorf("(etcd-bootstrap) member is still present in the etcd cluster membership")
```

Comment:
`IsBootstrapComplete()` is supposed to return true if complete and false if not. The error should only be returned if we have an actual error condition e.g failing to list members.

Finding that the etcd-bootstrap member is present is an expected state during bootstrapping and should not result in an error.
As it is, the usage of  `IsBootstrapComplete()` in the clustermemberremoval controller [L96-102](https://github.com/openshift/cluster-etcd-operator/blob/9cf91a0df3bfd9407b10811b5eb709c3160df371/pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go#L96-L102) will always result in the error `IsBootstrapComplete failed to determine bootstrap status: (etcd-bootstrap) member is still present in the etcd cluster membership` being returned during bootstrap and that controller will go degraded.
```Go
func (c *clusterMemberRemovalController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
	// skip reconciling this controller, if bootstrapping is not completed
	bootstrapComplete, err := ceohelpers.IsBootstrapComplete(c.configMapLister, c.operatorClient, c.etcdClient)
	if err != nil {
		return fmt.Errorf("IsBootstrapComplete failed to determine bootstrap status: %w", err)
	}
	if !bootstrapComplete {
		return nil
	}
```

Same for any other controller relying on this e.g etcdendpointscontroller
https://github.com/openshift/cluster-etcd-operator/blob/9cf91a0df3bfd9407b10811b5eb709c3160df371/pkg/operator/etcdendpointscontroller/etcdendpointscontroller.go#L97-L100

Reply:
Having a quick look around at the [past CI runs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-etcd-operator/1027/pull-ci-openshift-cluster-etcd-operator-master-e2e-aws-ovn-serial/1638564021026164736/artifacts/e2e-aws-ovn-serial/gather-extra/artifacts/pods/openshift-etcd-operator_etcd-operator-8b57fd5bc-t287x_etcd-operator.log) it doesn't seem like the controllers get degraded during bootstrap because `IsBootstrapComplete()` is correctly returning a `false, nil` from the other pre-checks earlier up.
However if for whatever reason we actually hit this condition of the member being present then those controllers will unnecessarily go degraded.
So this should also be returning `false, nil` like the above pre-checks when the bootstrap member is present.

And since we don't want to silently be spinning on this if the bootstrap member is not being removed for whatever reason, then we can add a debug log here to avoid being spammed.
```Go
klog.V(2).Infof("(etcd-bootstrap) member is still present in the etcd cluster membership")
```

Reply:
+1 for both reviews 👍🏽  

I will use debug level to avoid spamming the CEO logs, i think it is fair 👍🏽

Reply:
Sorry typo earlier, debug is `klog.V(4)` I believe.